pkg/cmd/server/api/v1/types.go

```diff
@@ -833,6 +833,21 @@ type LDAPSyncConfig struct {
 	// OpenShift Group names
 	LDAPGroupUIDToOpenShiftGroupNameMapping map[string]string `json:"groupUIDNameMapping"`
 
+	// TolerateMissingMembers determines the behavior of the LDAP sync job when missing user entries are
+	// encountered. If 'true', an LDAP query for users that doesn't find any will be tolerated and an only
+	// and error will be logged. If 'false', the LDAP sync job will fail if a query for users doesn't find
+	// any. The default value is 'false'. Misconfigured LDAP sync jobs with this flag set to 'true' can cause
+	// group membership to be removed, so it is recommended to use this flag with caution.
+	TolerateMissingMembers bool `json:"tolerateMissingMembers"`
+
+	// TolerateOutOfScopeMembers determines the behavior of the LDAP sync job when out-of-scope user entries
+	// are encountered. If 'true', an LDAP query for a user that falls outside of the base DN given for the all
+	// user query will be tolerated and only an error will be logged. If 'false', the LDAP sync job will fail
+	// if a user query would search outside of the base DN specified by the all user query. Misconfigured LDAP
+	// sync jobs with this flag set to 'true' can result in groups missing users, so it is recommended to use
+	// this flag with caution.
+	TolerateOutOfScopeMembers bool `json:"tolerateOutOfScopeMembers"`
```

Comment:
I think I would have expected error tolerance settings under the query config, more like
```
kind: LDAPSyncConfig
apiVersion: v1
url: ldap://LDAP_SERVICE_IP:389
insecure: true
rfc2307:
    groupsQuery:
        baseDN: "ou=incomplete,ou=groups,ou=rfc2307,dc=example,dc=com"
        scope: sub
        derefAliases: never
        filter: (objectclass=groupOfNames)
    groupUIDAttribute: dn
    groupNameAttributes: [ cn ]
    groupMembershipAttributes: [ member ]
    usersQuery:
        baseDN: "ou=people,ou=rfc2307,dc=example,dc=com"
        scope: sub
        derefAliases: never
        filter: (objectclass=inetOrgPerson)
        tolerateNotFoundError: true
        tolerateOutOfScopeError: true
    userUIDAttribute: dn
    userNameAttributes: [ mail ]
```

that makes it clearer which query the tolerance applies to, and makes the "member" portion implicit